### PR TITLE
implement jl_rec_backtrace stackwalk for more platform configurations

### PR DESCRIPTION
**Pull request by vtjnash**
This adds support for most of the default configuration on most common
platforms (combinations of glibc/Darwin + AArch64/ARM/x86/x86_64). It
was already fully supported for Windows x86 and x86_64.

I would add support here for musl (since it is trivial), but they explicitly prohibit doing so, via their header design, as an intentional choice, so I don't know how to auto-detect support for it in the build to work around that.

A simple demo is:
```julia
julia> for t in ccall(:jl_live_tasks, Any, ()); println(t); ccall(:jlbacktracet, Cvoid, (Any,), t); println(); end
```

### CODE REVIEW

File: src/stackwalk.c [hunk 1]
```diff
@@ -697,6 +697,161 @@ void jl_print_bt_entry_codeloc(jl_bt_element_t *bt_entry) JL_NOTSAFEPOINT
...
+    // 3. xor SPs to obtain key.
+    julia_longjmp_xor_key = mangled_sp ^ sp;
+}
+#endif
+
+JL_UNUSED static uintptr_t ptr_demangle(uintptr_t p)
+{
+#if defined(__GLIBC__)
+#if defined(_CPU_X86_)
+// from https://github.com/bminor/glibc/blame/master/sysdeps/unix/sysv/linux/i386/sysdep.h
```
Comment (giordano) [inline src/stackwalk.c:729]: It'd be nicer to use permalinks instead of referencing master

**Comment by jpsamaroo** [comment]
Is it possible to just assume that non-glibc means musl? It's most likely far more common than uClibc and other libc's. And if other libc's have a way to identify themselves, then assuming anything else is musl seems reasonably safe.

File: src/stackwalk.c (hunk 1 above)
Comment (vtjnash) [reply src/stackwalk.c:729]: these intentionally reference blame, since if the code changes any, I want that to be very obvious there to the reader

**Comment by vtjnash** [comment]
musl explicitly does not want to be given support in this (this PR is "badly wrong"). Unless they change their position, I am inclined not to support it therefore (https://www.openwall.com/lists/musl/2013/03/29/13)

**Comment by giordano** [comment]
This PR introduced a lot of new compilation warnings on aarch64-linux-gnu, including "outside array bounds": https://github.com/JuliaLang/julia/issues/45400#issuecomment-1159449826

**Pull request merged**